### ISSUE TITLE
'System.ApplicationException' in EmguNuGet.dll can't be caught

### ISSUE THREAD

**Issue by Szpoti** [emgucv/emgucv#516]
Hi there,

Im currently working on a small application, which bases to be yet implemented in a bigger software.
I told them to use EmguCV for webcamera reading, and we are using a different QRCode reader library to search for QR codes on the image retrived, and write it out if it does, that would be the whole point in the main software as well.
My problem is, sometimes the QRCode is read in, sometimes it's not, and in that specific case, I get the output
```Exception thrown: 'System.ApplicationException' in EmguNuGet.dll```
Afterwards, im unable to read any QRCode anymore, it just won't recognize it.
You can find the code and repo if intrested [here](https://github.com/Szpoti/EmguCVQrReadingTest/blob/main/Form1.cs), and a code snippet below:
```
private void button1_Click(object sender, EventArgs e)
        {
            if (Capture is null)
            {
                Capture = new VideoCapture();
                app = Start();
            }
            SetText("testing...");
            Image<Bgra, byte> image = Capture.QueryFrame().ToImage<Bgra, byte>();
            Capture.ImageGrabbed += Capture_ImageGrabbed;
            Capture.Start();
        }

private void Capture_ImageGrabbed(object sender, EventArgs e)
        {
            try
            {
                Mat m = new();
                Capture.Retrieve(m);
                Avalonia.PixelSize pixelSize = new(m.Size.Width, m.Size.Height);
                Avalonia.Vector dpi = new(96, 96);
                byte[,,] arr = (byte[,,])m.GetData();
                Avalonia.Platform.PixelFormat pixelFormat = Avalonia.Platform.PixelFormat.Rgba8888;
                Avalonia.Platform.AlphaFormat alphaFormat = Avalonia.Platform.AlphaFormat.Unpremul;
                var writeableBitmap = new WriteableBitmap(pixelSize, dpi, pixelFormat, alphaFormat);
                using (var fb = writeableBitmap.Lock())
                {
                    int[] data = new int[fb.Size.Width * fb.Size.Height];
                    for (int y = 0; y < fb.Size.Height; y++)
                    {
                        for (int x = 0; x < fb.Size.Width; x++)
                        {
                            byte r = arr[y, x, 0];
                            byte g = arr[y, x, 1];
                            byte b = arr[y, x, 2];

                            var color = new Color(255, r, g, b);

                            data[y * fb.Size.Width + x] = (int)color.ToUint32();
                        }
                    }
                    Marshal.Copy(data, 0, fb.Address, fb.Size.Width * fb.Size.Height);
                }

                var qrs = qRDecoder.SearchQrCodes(writeableBitmap);
                var q = qrs.Any();
                if (q)
                {
                    SetText(qrs.First());
                    qRDecoder = new QRDecoder();
                }
                pictureBox1.Image = m.ToBitmap();
            }
            catch (ApplicationException ex)
            {
                Capture.ImageGrabbed -= Capture_ImageGrabbed;
                Capture.Stop();
                Capture = null;
                app.Dispose();
                SetText(ex.Message);
            }
      }
```
I was trying to catch it but I can't because you guys already catching it I suppose. So I don't know what the problem could be.
Can you maybe help out where that exception could be coming from?
Thank you and looking forward for an answer

**Comment by emgucv** [comment]
Can you reproduce the bug with just a few line of Emgu CV only code? Please also attach a test image used by the code. 

Thanks.

**Comment by Szpoti** [comment]
I'm not quite sure what do you mean, but I was able to do this:
```
private void button1_Click(object sender, EventArgs e)
        {
            Capture = new VideoCapture();
            Capture.ImageGrabbed += Capture_ImageGrabbed;
            Capture.Start();
            detector = new();
        }

private void Capture_ImageGrabbed(object sender, EventArgs e)
        {
            try
            {
                Mat m = new();
                Mat o = new();
                Capture.Retrieve(m);
                var qrs = detector.Detect(m, o);
                if (qrs)
                {
                    string s = detector.Decode(m, o);
                    SetText(s);
                }
                pictureBox1.Image = m.ToBitmap();
            }
            catch (ApplicationException ex)
            {
                Capture.ImageGrabbed -= Capture_ImageGrabbed;
                Capture.Stop();
                Capture = null;
                //app.Dispose();
                SetText(ex.Message);
            }
       }
```
Using your QRCodeDetector, but it gives back
```
System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'
```
 When I call `detector.Decode(m, o);`

**Comment by nopara73** [comment]
@Szpoti Please create a new .NET Core GitHub repo and explain exactly step by step what you have to do in order to reproduce this bug. Something similar to this one: https://github.com/nopara73/SslStreamBug

**Comment by emgucv** [comment]
Fixed in this commit: b80e0b49d645339a9ac86d61dd8cd3126d4698db

Will be included in the up coming 4.5.2 release.

Before the offical release, as a temporary work around, you can pass an empty Mat as a 3rd parameter to the "Decode" function. That will make it work too. 

The bug is due to missing code to handle the 3rd optional parameter, making it a required parameter before the fix.

**Comment by Szpoti** [comment]
It's working this way, thank you for the quick reply.
